components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.v1/src/main/java/org/wso2/carbon/identity/api/server/userstore/v1/core/ServerUserStoreService.java

```diff
@@ -238,35 +238,24 @@ public UserStoreConfigurationsRes getUserStoreByDomainId(String domainId) {
     }
 
     /**
-     * Retrieve the meta of configured user store type.
+     * Retrieve the meta of user store type.
      *
      * @param typeId the user store type id.
-     * @param limit  items per page.
-     * @param offset to specify the offset param.
-     * @param filter to specify the filtering capabilities.
-     * @param sort   to specify the sorting order.
      * @return MetaUserStoreType.
      */
 
-    public List<MetaUserStoreType> getUserStoreManagerProperties(String typeId, Integer limit, Integer offset,
-                                                                 String filter, String sort) {
-
-        handleNotImplementedBehaviour(limit, offset, filter, sort);
+    public MetaUserStoreType getUserStoreManagerProperties(String typeId) {
 
         UserStoreConfigService userStoreConfigService = UserStoreConfigServiceHolder.getUserStoreConfigService();
+        Set<String> classNames;
         try {
-            UserStoreDTO[] userStoreDTOS = userStoreConfigService.getUserStores();
-            if (userStoreDTOS.length == 0) {
-                throw handleException(Response.Status.NOT_FOUND, UserStoreConstants.ErrorMessage.ERROR_CODE_NOT_FOUND);
-            }
-            List<UserStoreDTO> userStoresByTypeNameList = new ArrayList<>();
-            for (UserStoreDTO userStoreDTO : userStoreDTOS) {
-                if (Objects.equals(getUserStoreType(base64URLDecodeId(typeId)), userStoreDTO.getClassName())) {
-                    userStoresByTypeNameList.add(userStoreDTO);
-                }
+            classNames = userStoreConfigService.getAvailableUserStoreClasses();
+            if (classNames.contains(getUserStoreType(base64URLDecodeId(typeId)))) {
```

Comment:
can classNames become null here? if so we need to add a null check.

Reply:
addressed in f145414e186a1a82b6bc5e4cdd984d641d680ca8